output.tf

```diff
@@ -1,13 +1,13 @@
 output "bastion_vm_name" {
-  value = google_compute_instance.bastion[0].name
+  value = var.osd_gcp_private ? google_compute_instance.bastion[0].name : null
```

Comment:
nice, that's exactly that I was thinking about, not output the bastion value when the variable it's not defined and it's not generated.